frontend/cs/r1cs/api_assertions.go

```diff
@@ -187,7 +209,7 @@ func (builder *builder) mustBeLessOrEqCst(a expr.LinearExpression, bound big.Int
 			l = builder.Sub(l, aBits[i])
 
 			added = append(added, builder.cs.AddConstraint(builder.newR1C(l, aBits[i], builder.cstZero())))
-			builder.MarkBoolean(aBits[i].(expr.LinearExpression))
+			builder.MarkBoolean(aBits[i])
```

Comment:
I'm trying to understand why do we mark `aBits[i]` boolean at all? Imo marking something boolean is useful when we are going to use it later in a boolean operation and then we do not have to assert. But in this method this doesn't seem to happen anyway?